app/controllers/api/accounts_controller.rb

```diff
@@ -0,0 +1,65 @@
+class Api::AccountsController < ApplicationController
```

Comment:
Overall solid work. This controller has more going on in it than I like to see. Should probably move some of this out to a PORO. If things are working and tested well go with it for now, but I'd like to see some of this cleaned up before evals. Some notes to follow below...